accounts/backends.py

```diff
@@ -59,9 +59,10 @@ def authenticate(self, request, remote_user, tokens=True):
                 user.is_staff = False
                 user.is_superuser = False
 
-        # Update groups
+        # Update groups: first remove them from existing groups then add them to new groups
+        user.groups.clear()
         for group_name in remote_user["groups"]:
-            group, _ = Group.objects.get_or_create(name=group_name)
+            group, _ = Group.objects.get_or_create(name=f"platform_{group_name}")
             if group not in user.groups.all():
```

Comment:
I think we could technically simplify this line since it's now guaranteed the user won't be in this group, but I'm a bit hesitant to add something like that. What are your thoughts?

Reply:
Do you mean remove the if statement? I guess the if statement will always be true, so there's no point of keeping it. What makes you hesitant?